src/main/java/sx/blah/discord/handle/impl/obj/Category.java

```diff
@@ -0,0 +1,231 @@
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Discord4J is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Discord4J.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package sx.blah.discord.handle.impl.obj;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import sx.blah.discord.Discord4J;
+import sx.blah.discord.api.IDiscordClient;
+import sx.blah.discord.api.IShard;
+import sx.blah.discord.api.internal.DiscordClientImpl;
+import sx.blah.discord.api.internal.DiscordEndpoints;
+import sx.blah.discord.api.internal.DiscordUtils;
+import sx.blah.discord.api.internal.json.objects.OverwriteObject;
+import sx.blah.discord.api.internal.json.requests.ChannelEditRequest;
+import sx.blah.discord.handle.obj.*;
+import sx.blah.discord.util.LogMarkers;
+import sx.blah.discord.util.PermissionUtils;
+import sx.blah.discord.util.cache.Cache;
+import sx.blah.discord.util.cache.LongMap;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class Category implements ICategory {
+
+	private final DiscordClientImpl client;
+	private final IShard shard;
+	private volatile String name;
+	private final long id;
+	private final IGuild guild;
+	private volatile int position;
+	private volatile boolean isNSFW;
+
+	public final Cache<PermissionOverride> userOverrides;
+	public final Cache<PermissionOverride> roleOverrides;
+
+	public Category(IShard shard, String name, long id, IGuild guild, int position, boolean isNSFW, Cache<PermissionOverride> userOverrides, Cache<PermissionOverride> roleOverrides) {
+		this.shard = shard;
+		this.client = (DiscordClientImpl) shard.getClient();
+		this.name = name;
+		this.guild = guild;
+		this.position = position;
+		this.id = id;
+		this.isNSFW = isNSFW;
+		this.userOverrides = userOverrides;
+		this.roleOverrides = roleOverrides;
+	}
+
+	public void setName(String name) {
+		this.name = name;
+	}
+
+	public void setPosition(int position) {
+		this.position = position;
+	}
+
+	public void setNSFW(boolean isNSFW) {
+		this.isNSFW = isNSFW;
+	}
+
+	@Override
+	public void delete() {
+		PermissionUtils.requirePermissions(getModifiedPermissions(getClient().getOurUser()), EnumSet.of(Permissions.MANAGE_CHANNEL));
+
+		((DiscordClientImpl) client).REQUESTS.DELETE.makeRequest(DiscordEndpoints.CHANNELS+id);
+	}
+
+	@Override
+	public boolean isDeleted() {
+		return getClient().getCategoryByID(id) != this;
+	}
+
+	@Override
+	public IGuild getGuild() {
+		return guild;
+	}
+
+	@Override
+	public int getPosition() {
```

Comment:
This must take into account the fact that multiple categories can have the same position. Look at Channel for reference.

Reply:
Whenever a category is moved it sends an update with the new position automatically. Or am I mistaking something?

Reply:
No I mean the fact that multiple categories can have the same position. We return the display position for channels and should be consistent with categories.